### DISCUSSION TITLE
Memory Usage

### DISCUSSION THREAD

**Discussion by StepanGavrilov** [fkie-cad/cwe_checker#406] [Q&A]
Hello @Enkelmann!
I have a service that works together with cwe_checker. I run 5 cwe_checker processes in parallel and get this result from memory

![Снимок экрана 2023-04-07 в 13 08 54](https://user-images.githubusercontent.com/55409307/230591281-5fa24018-27d9-476e-b5b2-b129d37525fc.png)

Are there any optimization options?
- max_steps=20
- file size avg 2mb

**Comment by Enkelmann** [comment]
Currently, there are no memory optimization options implemented. I consider cases of excessive memory usage to be bugs, but these usually happen together with runtime explosion problems. Also, 10 GB RAM usage is still an acceptable amount of memory usage from my point of view.

One could optimize memory usage by storing some analysis data on the hard drive or by simply further restricting analysis exactness (which would need not yet implemented analysis options). However, at the moment there is simply not enough developer capacity to implement such features.
